src/isar/services/utilities/scheduling_utilities.py

```diff
@@ -24,28 +30,192 @@ class SchedulingUtilities:
+
+        Raises
+        ------
+        HTTPException 409 Conflict
+            If state machine is not idle and therefore can not start a new mission
+        """
+        is_state_machine_ready_to_receive_mission = state == States.Idle
+        if not is_state_machine_ready_to_receive_mission:
+            error_message = f"Conflict - Mission already in progress - State: {state}"
+            self.logger.warning(error_message)
+            raise HTTPException(status_code=HTTPStatus.CONFLICT, detail=error_message)
+
+        return is_state_machine_ready_to_receive_mission
 
     def start_mission(self, mission: Mission, initial_pose: Optional[Pose]) -> None:
-        self._send_command(
-            StartMissionMessage(mission=deepcopy(mission), initial_pose=initial_pose),
-            self.queues.start_mission,
-        )
+        """Start mission
+
+        Raises
+        ------
+        HTTTPException 408 Request timeout
+            If there is a timeout while communicating with the state machine
+        """
+        try:
+            self._send_command(
+                StartMissionMessage(
+                    mission=deepcopy(mission), initial_pose=initial_pose
+                ),
+                self.queues.start_mission,
+            )
+        except QueueTimeoutError:
+            error_message = "Timeout - Failed to start mission in ISAR"
+            self.logger.error(error_message)
+            raise HTTPException(
+                status_code=HTTPStatus.REQUEST_TIMEOUT, detail=error_message
+            )
+        self.logger.info("OK - Mission started in ISAR")
 
     def pause_mission(self) -> ControlMissionResponse:
-        return self._send_command(True, self.queues.pause_mission)
+        """Pause mission
+
+        Raises
+        ------
+        HTTTPException 408 Request timeout
+            If there is a timeout while communicating with the state machine
+        """
+        try:
+            return self._send_command(True, self.queues.pause_mission)
+        except QueueTimeoutError:
+            error_message = "Timeout - Failed to pause mission"
+            self.logger.error(error_message)
+            raise HTTPException(
+                status_code=HTTPStatus.REQUEST_TIMEOUT, detail=error_message
+            )
+        finally:
+            self.logger.info("OK - Mission successfully paused")
 
     def resume_mission(self) -> ControlMissionResponse:
-        return self._send_command(True, self.queues.resume_mission)
+        """Resume mission
+
+        Raises
+        ------
+        HTTTPException 408 Request timeout
+            If there is a timeout while communicating with the state machine
+        """
+        try:
+            return self._send_command(True, self.queues.resume_mission)
+        except QueueTimeoutError:
+            error_message = "Timeout - Failed to resume mission"
+            self.logger.error(error_message)
+            raise HTTPException(
```

Comment:
408 Request timeout implies a user error (all 400 codes do), I believe this should be a 500 internal server error as it has nothing to do with the user input

Reply:
Good catch, changed to 500 Internal server error